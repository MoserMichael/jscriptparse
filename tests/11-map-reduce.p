    # testing higher order functions: map and reduce
    a = [1, 2, 3, 4]
    b = map(a, def(x) { return x * x })
    
    println("b: {b[0]} {b[1]} {b[2]} {b[3]}")
    
    c = reduce(b, def(x, y) { return x + y }, 0)
    println("sum of squares: {c}")

    # map on dictionary arguments
    a={ 'Ernie': 3, 'Bert': 4, 'Cookie-Monster' : 5, 'GraphCount': 100 }
    res = map(a,def(k,v) { "key: {k} age: {v} " })
    println("mapped values: { join( res ) }")
