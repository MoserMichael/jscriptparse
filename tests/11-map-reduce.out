b: 1 4 9 16
sum of squares: 30

